### PR TITLE
Test and document extra field type support

### PR DESCRIPTION
**Pull request by albertoperdomo2**
## Summary
- Add comprehensive tests proving `extra` parameter supports nil, int, string, and map types with deterministic CBOR encoding
- Fix hash algorithm documentation (SHA-256 -> FNV-64a) in architecture.md
- Document `extra` field purpose and supported types to prevent LoRA cache pollution

## Test plan
- [x] Run new tests: `go test ./pkg/kvcache/kvblock/... -v -run TestHash_Extra`
- [x] Verify all existing tests pass: `go test ./pkg/kvcache/kvblock/... -v`
- [x] Confirm CBOR determinism, map key ordering, and type differentiation
- [x] Verify vLLM compatibility scenarios encode successfully

## Related Issues
- #45

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -117,9 +117,14 @@ sequenceDiagram
...
 
 * **Token Chunking**: Prompts are converted to tokens, which are then grouped into fixed-size chunks (default: 16).
-* **Hash Algorithm**: A chained hash is computed. Each block's key is the **lower 64 bits of a SHA-256 hash**, generated from the CBOR-encoded `[parentHash, tokenChunk, extraKeys]` tuple.
+* **Hash Algorithm**: A chained hash is computed. Each block's key is an **FNV-64a hash**, generated from the CBOR-encoded `[parentHash, tokenChunk, extra]` tuple.
 * **Initialization**: The hash chain starts with a configurable `HashSeed`. This value's source **must** align with the `PYTHONHASHSEED` environment variable in the vLLM pods to ensure hashes are consistent across the entire system.
-
+* **Extra Parameter**: The third component of the hash tuple enables cache differentiation:
+  - **nil** (default): Standard prompts without LoRA or multi-modal content
+  - **int**: LoRA adapter ID (e.g., 42)
+  - **string**: Medium identifier or adapter name (e.g., "gpu")
```
Comment (vMaroon) [inline docs/architecture.md]: Medium identifier does not participate in the hash - only what can affect the KV content can be involved.

**Comment by vMaroon** [comment]
@albertoperdomo2 can you verify commit signatures and mend? https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification
